os/binfmt/binfmt_unloadmodule.c

```diff
@@ -189,11 +189,15 @@ int unload_module(FAR struct binary_s *binp)
 			}
 
 #endif
+			for (int section_idx = 0; section_idx < BIN_MAX; section_idx++) {
+				binp->sections[section_idx] = NULL;
+			}
 		}
 #else
 		/* Whole loading sections are in one memory block, so free the first allocated memory is enough. */
 		binfo("Freeing : %p\n", binp->sections[0]);
 		kmm_free((FAR void *)binp->sections[0]);
+		binp->sections[0] = NULL;
```

Comment:
Why is this assignment required? Isn't kmm_free enough?

Reply:
We seem to be freeing these section multiple times based on whether the section pointer is non zero. Even though this wont create any issue, its better to always set the pointer to null after freeing it to avoid any future possibility of dangling pointers.